xarray/tests/test_plot.py

```diff
@@ -1888,6 +1898,21 @@ def test_interval_breaks_logspace(self) -> None:
 class TestImshow(Common2dMixin, PlotTestCase):
     plotfunc = staticmethod(xplt.imshow)
 
+    @pytest.mark.xfail(
+        reason="Failing, should work. remove when works, already in Common2dMixin"
```

Comment:
Is this PR already ready? If so, why are these failing? Can you add a better reason?

Reply:
They have always failed, the old inherited tests only did `a.plot(x="t")`. 

Try on main:
```python
import pandas as pd
import xarray as xr
import numpy as np

time = pd.date_range("2000-01-01", "2000-01-10")
a = DataArray(np.random.randn(2, len(time)), [("xx", [1, 2]), ("t", time)])
a.plot.imshow(x="t") # Fails inside matplotlib :(

a.plot.contourf(x="t") # Works! :)
```

Reply:
Clarified the reason a little. Is it any better?

Reply:
thanks! Didn't realize that they were failing before, sry.